Dt/refactor client fetch

doubt72 opened the pull request:
Refactor to reduce double client fetch.

Commit:
First pass refactor to eliminate client double-fetch

Commit:
White space cleanup

Commit:
Refactoring to get 404 handling out of chef_wm_base

Commit:
Added not_found to client_state record to make dialyzer happy

seth commented on src/chef_wm_named_client.erl:
```diff
@@ -48,39 +48,45 @@ request_type() ->
+    {Req1, State1} = validate_any_request(Req, State),
+    #base_state{resource_state =
+                    #client_state{chef_client =
+                                      #chef_client{name = Name} = OldClient} =
```
I think this will crash if chef_client is set to not_found. So I think we want to have a case statement that does less work for the not_found case.

Commit:
Fix for crash when PUTing to a non-existent user

seth commented on src/chef_wm_named_client.erl:
```diff
@@ -48,39 +48,50 @@ request_type() ->
+        not_found ->
+            {Req1, State1};
+        _ ->
+            #chef_client{name = Name} = OldClient,
```
Is this Name ever different from the URL name? I think the answer is that the URL name and this name from the current/old client will always be the same. I don't have a preference I don't think, but might be worth a comment. Note also you can use the following syntax if you are extracting a single field from a record:

```
Name = OldClient#chef_client.name
```

seth commented:
Overall, this looks good to me. If you're up for adding a comment and think the alternative record syntax is helpful, do that and then merge.

Commit:
Added explanatory comment regarding request/old client names

They are the same; there can probably be a simplifying refactor of
clients in chef_object, but it's too big of a yak shave to be worth
doing at this exact moment
